x-pack/legacy/plugins/uptime/public/state/reducers/monitor.ts

```diff
@@ -42,10 +50,28 @@ export function monitorReducer(state = initialState, action: MonitorActionTypes)
         loading: false,
       };
     case FETCH_MONITOR_DETAILS_FAIL:
-      const error = action.payload;
       return {
         ...state,
-        errors: [...state.errors, error],
+        errors: [...state.errors, action.payload],
+      };
+    case FETCH_MONITOR_LOCATIONS:
+      return {
+        ...state,
+        loading: true,
+      };
+    case FETCH_MONITOR_LOCATIONS_SUCCESS:
+      const { monitorId: monId } = action.payload;
```

Comment:
Might it be cleaner just to reference the field directly, rather than aliasing it after the first reference to it? Maybe this is a little nitpicky, but this is something that will be a common problem that could yield lots of code with little aliases. For example I could add a new case below this one and not be able to re-use `monId` because it's declared in the same block scope already.

We could also destructure outside the switch to a type like `string | null` in this case. WDYT?

Reply:
i agree, i have done that.